haystack/document_stores/elasticsearch.py

```diff
@@ -301,6 +301,10 @@ def _create_document_index(self, index_name: str, headers: Optional[Dict[str, st
                     }
                 }
             }
+            if self.search_fields:
+                for field in self.search_fields:
+                    mapping["mappings"]["properties"].update({field: {"type": "text"}})
+
             if self.synonyms:
                 mapping["mappings"]["properties"][self.content_field] = {"type": "text", "analyzer": "synonym"}
                 mapping["settings"]["analysis"]["analyzer"]["synonym"] = {"tokenizer": "whitespace",
```

Comment:
Please also set the search_fields types to text if synonyms have been provided.

Reply:
Do you mean to add the synonym analyzer for the search field as well if the synonyms are passed

Reply:
Yes, functionally that's ok. However now we're passing always the same dictionary as field mapping. This can be confusing if later on someone tries to change the mapping for one field only but ends up with changing all `self.search_fields` mappings.

So here's what I recommend: 
- [ ] get rid of the extra handling for self.content_field in https://github.com/deepset-ai/haystack/blob/51d15e381541dcc986775c9c706ae57bbdb813cc/haystack/document_stores/elasticsearch.py#L284 and https://github.com/deepset-ai/haystack/blob/51d15e381541dcc986775c9c706ae57bbdb813cc/haystack/document_stores/elasticsearch.py#L314 This is per default included in self.search_fields, anyway.
- [ ] instead of line 314 loop over `self.search_fields` and add the mapping with analyzer
- [ ] add an else block after synonyms handling: set the mapping without analyzer for all `self.search_fields`
- [ ] and there's another thing missing I totally forgot: we have to do the same for `OpenSearchDocumentStore`, so copy anything from Line 280 till the end of the else block of the last bullet point and replace https://github.com/deepset-ai/haystack/blob/51d15e381541dcc986775c9c706ae57bbdb813cc/haystack/document_stores/elasticsearch.py#L1361-L1384